client/ayon_core/pipeline/actions.py

```diff
@@ -10,6 +13,224 @@
+        task_id,
+        folder_path=None,
+        task_name=None,
+        project_entity=None,
+        folder_entity=None,
+        task_entity=None
+    ):
+        self._project_name = project_name
+        self._folder_id = folder_id
+        self._task_id = task_id
+
+        self._folder_path = folder_path
+        self._task_name = task_name
+
+        self._project_entity = project_entity
+        self._folder_entity = folder_entity
+        self._task_entity = task_entity
+
+    def __getitem__(self, key):
+        if key in {"project_name", "AYON_PROJECT_NAME", "AVALON_PROJECT"}:
+            return self.project_name
+        if key == {"folder_path", "AYON_FOLDER_PATH", "AVALON_ASSET"}:
+            return self.folder_path
+        if key == {"task_name", "AYON_TASK_NAME", "AVALON_TASK"}:
+            return self.task_name
+        if key == "folder_id":
+            return self.folder_id
+        if key == "task_id":
+            return self.task_id
+        if key == "project_entity":
+            return self.project_entity
+        if key == "folder_entity":
+            return self.folder_entity
+        if key == "task_entity":
+            return self.task_entity
+        raise KeyError(f"Key: {key} not found")
+
+    def __contains__(self, key):
+        # Fake missing keys check for backwards compatibility
+        if key in {
+            "AYON_PROJECT_NAME",
+            "AVALON_PROJECT",
+            "project_entity",
+        }:
+            return self._project_name is not None
+        if key in {
+            "AYON_FOLDER_PATH",
+            "folder_id",
+            "folder_path",
+            "folder_entity",
+            "AVALON_ASSET",
+        }:
+            return self._folder_id is not None
+        if key in {
+            "AYON_TASK_NAME",
+            "task_id",
+            "task_name",
+            "task_entity",
+            "AVALON_TASK",
+        }:
+            return self._task_id is not None
+        return False
+
+    def get(self, key, default=None):
+        try:
+            return self[key]
+        except KeyError:
+            return default
+
+    def get_project_name(self):
+        """Selected project name.
+
+        Returns:
+            Union[str, None]: Selected project name.
+
+        """
+        return self._project_name
+
+    def get_folder_id(self):
+        """Selected folder id.
+
+        Returns:
+            Union[str, None]: Selected folder id.
+
+        """
+        return self._folder_id
+
+    def get_folder_path(self):
+        """Selected folder path.
+
+        Returns:
+            Union[str, None]: Selected folder path.
+
+        """
+        if self._folder_id is None:
+            return None
+        if self._folder_path is None:
+            self._folder_path = self.folder_entity["path"]
+        return self._folder_path
+
+    def get_task_id(self):
+        """Selected task id.
+
+        Returns:
+            Union[str, None]: Selected task id.
```

Comment:
Isn't optional clearer here?
```suggestion
            Optional[str]: Selected task id.
```

Reply:
To be honest, I don't like using `Optional` in return type hint.